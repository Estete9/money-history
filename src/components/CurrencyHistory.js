import { useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import SectionHeader from './SectionHeader';
import { fetchCurrencyHistory } from '../redux/currencies/currenciesSlice';

function CurrencyHistory() {
  const { symbol } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    const currentDate = new Date();
    const getPast5Years = () => {
      const years = [];
      let currentYear = currentDate.getFullYear();
      for (let i = 0; i < 5; i += 1) {
        years.push(currentYear - 1);
        currentYear -= 1;
      }
      return years;
    };
    const past5Years = getPast5Years();
    past5Years.map((year) => dispatch(fetchCurrencyHistory({ symbol, currentDate, year })));
  }, [dispatch, symbol]);

  return (
    <div>
      <SectionHeader name={symbol} value="1.594" />
      <h1>{`this is the symbol passed ${symbol}`}</h1>
    </div>
  );
}

export default CurrencyHistory;
