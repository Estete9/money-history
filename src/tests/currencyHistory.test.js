import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
// import axios from 'axios';
import { Provider } from 'react-redux';
import CurrencyHistory from '../components/CurrencyHistory';

jest.mock('axios');

describe('Integration tests for CurrenciesHistory component', () => {
  const mockConfigStore = configureStore([]);
  test('Checks if loading screen in CurrencyHistory renders correctly', () => {
    const historyStore = mockConfigStore({
      currencies: {
        currencyHistory: [],
        isLoadingHistory: true,
      },
    });
    const component = renderer.create(
      <Provider store={historyStore}>
        <CurrencyHistory />
      </Provider>,
    );
    const currencyHistoryTree = component.toJSON();

    expect(currencyHistoryTree).toMatchSnapshot();
  });
});
