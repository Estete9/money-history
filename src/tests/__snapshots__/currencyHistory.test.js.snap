// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests for CurrenciesHistory component Checks if loading screen in CurrencyHistory renders correctly 1`] = `
<div>
  Currency's History loading...
</div>
`;
